### ISSUE TITLE
Support for Basic Auth

### ISSUE THREAD

**Issue by tareksamni** [brotandgames/ciao#19]
Hi @brotandgames, I suggest to natively support Basic Auth using env vars. So for example, if `BASIC_AUTH_USERNAME` is defined then we enable basic auth authentication otherwise the application will be publicly available.

Something like:
```
class ApplicationController < ActionController::Base
  before_filter :authenticate
  protect_from_forgery unless: -> { request.format.json? }

  def authenticate
    basic_auth_username = ENV.fetch('BASIC_AUTH_USERNAME', '')
    basic_auth_password = ENV.fetch('BASIC_AUTH_PASSWORD', '')

    return true if basic_auth_username.empty?

    authenticate_or_request_with_http_basic('Ciao Application') do |username, password|
      username == basic_auth_username && password == basic_auth_password
    end
  end
end
```

Should be sufficient, simple and nice :) what do you think?

**Comment by brotandgames** [comment]
@tareksamni 
> Should be sufficient, simple and nice :) what do you think?

Yes, looks good :) Thank you for your proposals and contributions!

**Comment by brotandgames** [comment]
Closed by #21.
